src/conn/conn_dhandle.c

```diff
@@ -314,8 +314,6 @@ __wt_conn_btree_open(
 	    F_ISSET(dhandle, WT_DHANDLE_EXCLUSIVE) &&
 	    !LF_ISSET(WT_DHANDLE_LOCK_ONLY));
 
-	WT_ASSERT(session, !F_ISSET(S2C(session), WT_CONN_CLOSING));
```

Comment:
> `WT_ASSERT(session, !F_ISSET(S2C(session), WT_CONN_CLOSING));`
> What is the motivation for removing this assertion?

I've restored this assertion in b7e94da, using a new flag, `WT_CONN_CLOSING_NO_MORE_OPENS`.